File: src/java/magma/Node.java
package magma;

import java.util.List;
import java.util.Optional;

public interface Node {
    Node withNodeList(String propertyKey, List<Node> propertyValues);

    Optional<List<Node>> findNodeList(String propertyKey);

    Node withString(String propertyKey, String propertyValue);

    Optional<String> findString(String propertyKey);

    Node withNode(String propertyKey, Node propertyValue);

    Optional<Node> findNode(String propertyKey);
}

File: src/c/magma/Node.c
#include <temp.h>
#include <temp.h>
struct Node {
	struct Node new(){
		struct Node this;
		return this;
	}
	Node withNodeList(void* _this_);
	Optional<List<Node>> findNodeList(void* _this_);
	Node withString(void* _this_);
	Optional<String> findString(void* _this_);
	Node withNode(void* _this_);
	Optional<Node> findNode(void* _this_);
};